samples/nrf9160/coap_client/README.rst

```diff
@@ -3,12 +3,24 @@
 nRF9160: nRF CoAP Client
 ########################
 
-The nRF CoAP Client sample demonstrates how to receive data from a public CoAP server with an nRF9160 SiP.
+The nRF CoAP Client sample demonstrates the communication between a public CoAP server and an nRF9160 SiP which acts as the CoAP client.
 
 Overview
-*********
+********
 
-The sample connects to a public CoAP test server, sends periodic GET request for a test resource that is available on the server, and prints the data that is received.
+The nRF CoAP Client sample performs the following actions:
+
+#. Connect to the configured public CoAP test server (indicated by the parameter ``CONFIG_COAP_SERVER_HOSTNAME``)
+#. Send periodic GET request for a test resource (indicated by the parameter ``CONFIG_COAP_RESOURCE``) that is available on the server
+#. Display the received data about the resource on a terminal emulator.
+
+The public CoAP server used in this sample is Californium CoAP server (coap://californium.eclipse.org:5683).
+This server runs Eclipse Californium which is an open source implementation of the CoAP protocol that is targeted at the development and testing of IoT applications.
+An nRF9160 DK board is used as the CoAP client.
+
+An observable resource **obs** (Californium observable resource) is used in the communication between the CoAP client and the public CoAP server.
+The communication follows the common observe pattern and it defines two roles - an observer and an observable subject.
```

Comment:
This is not correct, the sampel does not implement CoAP observe mechanism, just simple quering.

I guess the confusion might come from the default resource the sample queries (`obs`), but the truth is, that was the only resource I found that provided any reasonable data (current time). There's no documentation for this server, so it had to be inspected "manually".

Reply:
Modified the section as : 
This sample uses the resource **obs** (Californium observable resource) in the communication between the CoAP client and the public CoAP server.
The communication follows the standard request/response pattern.
The sample queries one resource at a time.
Other resources can be configured using the Kconfig option ``CONFIG_COAP_RESOURCE``.
The CoAP client is notified of any changes in the state of the value of the resource from the server.

Reply:
> The CoAP client is notified of any changes in the state of the value of the resource from the server.

This would be correct if we have used the observe mechanism, but we don't. In communication based on requests/responses, there's no proactive communication from the server-side.

Reply:
How about modifying as follows: 
This sample uses the resource **obs** (Californium observable resource) in the communication between the CoAP client and the public CoAP server.
The communication follows the standard request/response pattern and is based on the change in the state of the value of the resource.
The sample queries one resource at a time.
Other resources can be configured using the Kconfig option ``CONFIG_COAP_RESOURCE``.

Reply:
Not sure about the `is based on` part, but I'm not the language expert so perhaps it's fine, thanks!